From 7115dbd41f10aa34e32092ddf5e2affe8040695c Mon Sep 17 00:00:00 2001
From: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
Date: Mon, 7 Sep 2009 09:26:30 +0900
Subject: [PATCH 02/12] USB: EMMA EHCI/OHCI: Remove emma_flip_init

FLIP_M0_CTR and FLIP_USB_CTR are common registers for built-in USB HCs,
so we'd better to initialize them in one place, in most cases, at
platform initialization part.

Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
---
 drivers/usb/host/ehci-emma.c |   29 -----------------------------
 drivers/usb/host/ohci-emma.c |   29 -----------------------------
 2 files changed, 0 insertions(+), 58 deletions(-)

diff --git a/drivers/usb/host/ehci-emma.c b/drivers/usb/host/ehci-emma.c
index 9b699aa..fa0673d 100644
--- a/drivers/usb/host/ehci-emma.c
+++ b/drivers/usb/host/ehci-emma.c
@@ -28,21 +28,6 @@
 #include <linux/usb.h>
 #include <asm/emma/emmaxxx.h>
 
-/* ftop registers */
-#define EMMA_FLIP_REGBASE	(0xb0200100)
-#define EMMA_FLIP_M0_CTR	(0x0)
-#define EMMA_FLIP_USB_CTR	(0x20)
-
-#define FLIP_OCI1_EN		(0x00100000UL)
-#define FLIP_OCI2_EN		(0x00200000UL)
-#define FLIP_PPON1_EN		(0x00400000UL)
-#define FLIP_PPON2_EN		(0x00800000UL)
-#define FLIP_OCI1_ACT		(0x00010000UL)
-#define FLIP_OCI2_ACT		(0x00020000UL)
-#define FLIP_PPON1_ACT		(0x00040000UL)
-#define FLIP_PPON2_ACT		(0x00080000UL)
-#define FLIP_ENDIAN_SWAP	(0x00000002UL)
-
 /* ftop pseudo pci */
 #define EHCI_CONFBASE		(0xb0200500UL)
 #define OHCI_CONFBASE		(0xb0200200UL)
@@ -50,19 +35,6 @@
 #define	PCI_CMD_MEM_ENABLE	(0x0002)
 #define	PCI_CMD_MASTER_ENABLE	(0x0004)
 
-static int emma_flip_init(void)
-{
-	u32	d;
-
-	writel(0x00ff0900, (void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_USB_CTR));
-
-	d = readl((void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_M0_CTR));
-	d |= FLIP_ENDIAN_SWAP;
-	writel(d, (void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_M0_CTR));
-
-	return 0;
-}
-
 static void emma_pseudo_pci_init(struct platform_device *pdev)
 {
 	struct resource *res;
@@ -90,7 +62,6 @@ static void emma_start_ehc(struct platform_device *pdev)
 {
 	pr_debug(__FILE__ ": starting EMMA EHCI USB Controller\n");
 
-	emma_flip_init();
 	emma_pseudo_pci_init(pdev);
 }
 
diff --git a/drivers/usb/host/ohci-emma.c b/drivers/usb/host/ohci-emma.c
index 686c603..77ad987 100644
--- a/drivers/usb/host/ohci-emma.c
+++ b/drivers/usb/host/ohci-emma.c
@@ -27,21 +27,6 @@
 #include <linux/usb.h>
 #include <asm/emma/emmaxxx.h>
 
-/* ftop registers */
-#define EMMA_FLIP_REGBASE	(0xb0200100)
-#define EMMA_FLIP_M0_CTR	(0x0)
-#define EMMA_FLIP_USB_CTR	(0x20)
-
-#define FLIP_OCI1_EN		(0x00100000UL)
-#define FLIP_OCI2_EN		(0x00200000UL)
-#define FLIP_PPON1_EN		(0x00400000UL)
-#define FLIP_PPON2_EN		(0x00800000UL)
-#define FLIP_OCI1_ACT		(0x00010000UL)
-#define FLIP_OCI2_ACT		(0x00020000UL)
-#define FLIP_PPON1_ACT		(0x00040000UL)
-#define FLIP_PPON2_ACT		(0x00080000UL)
-#define FLIP_ENDIAN_SWAP	(0x00000002UL)
-
 /* ftop pseudo pci */
 #define EHCI_CONFBASE		(0xb0200500UL)
 #define OHCI_CONFBASE		(0xb0200200UL)
@@ -49,19 +34,6 @@
 #define	PCI_CMD_MEM_ENABLE	(0x0002)
 #define	PCI_CMD_MASTER_ENABLE	(0x0004)
 
-static int emma_flip_init(void)
-{
-	u32	d;
-
-	writel(0x00ff0900, (void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_USB_CTR));
-
-	d = readl((void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_M0_CTR));
-	d |= FLIP_ENDIAN_SWAP;
-	writel(d, (void *)(EMMA_FLIP_REGBASE + EMMA_FLIP_M0_CTR));
-
-	return 0;
-}
-
 static void emma_pseudo_pci_init(struct platform_device *pdev)
 {
 	struct resource *res;
@@ -89,7 +61,6 @@ static void emma_start_hc(struct platform_device *pdev)
 {
 	pr_debug(__FILE__ ": starting EMMA OHCI USB Controller\n");
 
-	emma_flip_init();
 	emma_pseudo_pci_init(pdev);
 }
 
-- 
1.6.3.3

